EJERCICIO-PRACTICA-2-LIBRERIA-RIOS TAPIA BERNARDO .cpp

```diff
@@ -0,0 +1,478 @@
+                int eid1 = e1 ? e1->getId() : -1;
+                int eid2 = e2 ? e2->getId() : -1;
+                if (id1 == eid1 || id1 == eid2 || id2 == eid1 || id2 == eid2) {
+                    cout << "Regla: un libro no puede repetirse en mas de un kit dentro del mismo pedido" << endl;
+                    return false;
+                }
+            }
+        }
+
+        kits[cantidad] = k;
+        cantidad = cantidad + 1;
+        return true;
+    }
+
+    void mostrarPedido() {
+        cout << "Pedido[ maxKits=" << maxNumeroKits << ", cantidad=" << cantidad << " ]" << endl;
+        for (int i = 0; i < cantidad; i++) {
+            if (kits[i] != 0) {
+                cout << "  ";
+                kits[i]->mostrarKit();
+            }
+        }
+    }
+};
+
+//ListaPedidos.h
+#pragma once
+#include <iostream>
+#include "Pedido.h"
+using namespace std;
+
+class ListaPedidos {
+private:
+    Pedido** listaPedidos;
+    int capacidad;
+    int cantidad;
+
+public:
+    ListaPedidos(int capacidad) {
+        this->capacidad = capacidad;
+        this->cantidad = 0;
+        listaPedidos = new Pedido*[capacidad];
+        for (int i = 0; i < capacidad; i++) {
+            listaPedidos[i] = 0;
+        }
+    }
+
+    ~ListaPedidos() {
+        for (int i = 0; i < cantidad; i++) {
+            delete listaPedidos[i];
+            listaPedidos[i] = 0;
+        }
+        delete[] listaPedidos;
+        listaPedidos = 0;
+    }
+
+    void registrarPedido(Pedido* p) {
+        if (p == 0) {
+            return;
+        }
+        if (cantidad < capacidad) {
+            listaPedidos[cantidad] = p;
+            cantidad = cantidad + 1;
+        } else {
+            cout << "No se pueden agregar mas pedidos (capacidad llena)" << endl;
+            delete p;
+        }
+    }
+
+    void mostrarPedidos() {
+        for (int i = 0; i < cantidad; i++) {
+            if (listaPedidos[i] != 0) {
+                listaPedidos[i]->mostrarPedido();
+            }
+        }
+    }
+};
+
+//Losta.h
+
+#pragma once
+#include "ListaLibros.h"
+#include "ListaKits.h"
+#include "ListaPedidos.h"
+
+class Libreria {
+private:
+    ListaLibros* listaLibros;
+    ListaKits* listaKits;
+    ListaPedidos* listaPedidos;
+
+public:
+    Libreria(int capLibros, int capKits, int capPedidos) {
+        listaLibros = new ListaLibros(capLibros);
+        listaKits = new ListaKits(capKits);
+        listaPedidos = new ListaPedidos(capPedidos);
+    }
+
+    ~Libreria() {
+        delete listaLibros;
+        delete listaKits;
+        delete listaPedidos;
+        listaLibros = 0;
+        listaKits = 0;
+        listaPedidos = 0;
+    }
+
+    // Libros
+    void registrarLibro(Libro* libro) {
+        listaLibros->registrarLibro(libro);
+    }
+
+    void mostrarLibros() {
+        listaLibros->mostrarLibros();
+    }
+
+    // Kits
+    void crearYRegistrarKit(string nombreKit, int precio, int idLibro1, int idLibro2) {
```

Comment:
Debe recibir como parametro el objeto kit , 
```suggestion
    void registrarKit(Kit* nuevoKit) {
```